package.json

```diff
@@ -182,6 +182,14 @@
     "react": "^16.8.6",
     "react-dom": "^16.8.6"
   },
+  "peerDependenciesMeta ": {
+    "piwik-react-router": {
+      "optional": true
+    },
```

Comment:
you can't make piwik optional. The package is used inside UI itself: https://github.com/cozy/cozy-ui/blob/3826219cc24ea4182a8ea57f7d8cf14ff9ffb691/react/helpers/tracker.jsx#L45

Reply:
piwik will stay a peer Dependency.

Quoting the doc: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta
> Marking a peer dependency as optional ensures npm will not emit a warning if the soy-milk package is not installed on the host. This allows you to integrate and interact with a variety of host packages without requiring all of them to be installed.

Reply:
As it's not mandatory to be using tracker.jsx inside app, I think we should not be warning during install of apps that don't have piwik-react-router in dependencies.

Reply:
I think this lib is mandatory, since all our apps are using getTracker(). I don't know why it was put in peerDep at first, maybe because of version's conflict but this doesn't reflect our internal guidelines, right?

Reply:
https://github.com/cozy/cozy-ui/commit/e3bd229a9b1b8a68aa760c078adff80ab61dab3d

@JF-Cozy maybe?

Reply:
First adding was strange : https://github.com/cozy/cozy-ui/commit/77274a6631e8c97d15f785145dbdb0420545cad7

PeerDependencies should not impact yarn.lock, in my understanding.

Reply:
Let's follow our guideline:

piwik-react-router should be part of our "dependencies"

Reply:
> PeerDependencies should not impact yarn.lock, in my understanding.

Since the lockfile is there to fix versions, it's legit to have the peerDep listed in it.

Reply:
> Since the lockfile is there to fix versions, it's legit to have the peerDep listed in it.

In order to add version in the lockfile, the package should be set in the dependencies or devDependencies array.

That is why @JF-Cozy fix the warning (because time and time, the presence inside the yarnlock was removed, I guess), by adding the piwik-react-router as devDependency.

Reply:
>by adding the piwik-react-router as devDependency.

Since you need the dep to dev / test the feature, it's legit to have it in devDep. 

ok for the lock file

Reply:
new version: 
- Define puppeteer as optional peer dependencies 
- piwik-react-router as dependencies